src/main/resources/assets/carpet/scripts/draw.sc

```diff
@@ -0,0 +1,196 @@
+        set([x, y, z],block);
+        success+=1;
+
+        //todo add check for carpet setting CarpetSettings.fillUpdates
+        update(block(x, y, z))
+    );
+    success
+);
+
+length_sq(vec)->return(_vec_length(vec)^2);//cos of lengthSq func in DrawCommand.java
+
+fill_flat(pos, offset, dr, rectangle, orientation, block, hollow, replacement)->(
+    successes = 0;
+    r = floor(dr);
+    drsq = dr^2;
+    if(orientation=='x',
+        c_for(a=-r,a<=r,a+=1,
+            c_for(b=-r,b<=r,b+=1,
+                if((!hollow && (rectangle || a*a + b*b <= drsq))||//if not hollow, vry simple
+                    (hollow && ((rectangle && (abs(a) == r || abs(b) ==r)) || //If hollow and it's a rectangle
+                    (!rectangle && (a*a + b*b <= drsq && (abs(a)+1)^ 2 + (abs(b)+1)^2 >= drsq)))),//If hollow and not rectangle
+                    successes+=set_block(pos:0+offset,pos:1+a,pos:2+b,block, replacement)
+                )
+            )
+        ),
+        orientation == 'y',
+        c_for(a=-r,a<=r,a+=1,
+            c_for(b=-r,b<=r,b+=1,
+                if((!hollow && (rectangle || a*a + b*b <= drsq))||//if not hollow, vry simple
+                    (hollow && ((rectangle && (abs(a) == r || abs(b) ==r)) || //If hollow and it's a rectangle
+                    (!rectangle && (a*a + b*b <= drsq && (abs(a)+1)^ 2 + (abs(b)+1)^2 >= drsq)))),//If hollow and not rectangle
+                    successes+=set_block(pos:0+a,pos:1+offset,pos:2+b,block, replacement)
+                )
+            )
+        ),
+        orientation == 'z',
+        c_for(a=-r,a<=r,a+=1,
+            c_for(b=-r,b<=r,b+=1,
+                if((!hollow && (rectangle || a*a + b*b <= drsq))||//if not hollow, vry simple
+                    (hollow && ((rectangle && (abs(a) == r || abs(b) ==r)) || //If hollow and it's a rectangle
+                    (!rectangle && (a*a + b*b <= drsq && (abs(a)+1)^2 + (abs(b)+1)^2 >= drsq)))),//If hollow and not rectangle
+                    successes+=set_block(pos:0+b,pos:1+a,pos:2+offset,block, replacement)
+                )
+            )
+        ),
+        print(player(),format('r Error while running command: orientation can only be "x", "y" or "z", '+orientation+' is invalid.'))
+    );
+    successes
+);
+
+//drawing commands
+draw_sphere(pos, radius, block, replacement, hollow)->(
+    affected = 0;
+
+    radiusX = radius+0.5;
+    radiusY = radius+0.5;
+    radiusZ = radius+0.5;
+
+    ceilRadiusX = ceil(radiusX);
+    ceilRadiusY = ceil(radiusY);
+    ceilRadiusZ = ceil(radiusZ);
+
+    nextXn = 0;
+
+    c_for(x=0,x<=ceilRadiusX,x+=1,
+        xn=nextXn;
+        nextXn = (x+1)/radiusX;
+        nextYn = 0;
+        c_for(y=0,y<=ceilRadiusY,y+=1,
+            yn = nextYn;
+            nextYn = (y + 1) / radiusY;
+            nextZn = 0;
+            c_for(z=0,z<=ceilRadiusZ,z+=1,
+                zn = nextZn;
+                nextZn = (z+1)/radiusZ;
+
+                if(length_sq([xn, yn, zn]) > 1,break());
```

Comment:
try not to use breaks / continues especially in the innermost loop

Reply:
First of all, u did that (copied directly from ur code), second of all, I little other way to do it.

Reply:
AFAIK, it's because in Scarpet `return` calls (and probably `break` calls too) are expensive.

Reply:
I get that, but that doesn't change the fact that I see little way around it.

Reply:
I'm open to suggestions

Reply:
Maybe wrapping everything into some big `if` statements and do nothing if the condition isn't met, for the break (if I'm not mistaken and breaks exits the entire loop) maybe set a variable that is reset outside the loop to true and check in the same `if`. Don't know if that's actually better or worse (didn't test it), but just an idea.

Reply:
Wow you can only really use `break()` 750 times a tick in my computer, same for `continue()`.

Empty `if` can, however, be ran 107k times.

Reply:
oof wow Imma change that

Reply:
Did it and made a PR to your repo at Ghoulboy78#19